src/openforms/appointments/contrib/qmatic/plugin.py

```diff
@@ -0,0 +1,175 @@
+import json
+import logging
+from datetime import date, datetime, time
+
+from openforms.appointments.contrib.qmatic.models import QmaticConfig
+
+from ..base import (
+    AppointmentClient,
+    AppointmentLocation,
+    AppointmentProduct,
+    BasePlugin,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Plugin(BasePlugin):
+    """
+    Plugin for Qmatic Orchestra Calendar Public Appointment API (july 2017)
+    """
+
+    def __init__(self):
+        config = QmaticConfig.get_solo()
+        if not config.service:
+            logger.warning("No service defined for Qmatic")
+            return
+
+        self.client = config.service.build_client()
+        self.api_root = config.service.api_root
```

Comment:
I'd move this to a `def client`, decorated with `@property` to make it lazy. On object initialization, you usually don't want to do blocking calls (which the DB query is). This is especially relevant when initializing the registry at code-import time where plugins are initialized, which runs before the DB is even migrated on fresh installs.

Reply:
Done slightly different but does the job.